dotCMS/src/main/java/com/dotcms/rest/api/v1/system/monitor/MonitorHelper.java

```diff
@@ -96,125 +73,109 @@ class MonitorHelper {
                     }
                 }
             }
-        }catch(Exception e){
+        } catch (Exception e) {
             Logger.warnAndDebug(this.getClass(), e.getMessage(), e);
             throw new DotRuntimeException(e);
         }
     }
 
-    
-    static Tuple2<Long,MonitorStats> cachedStats=null;
-    
-    MonitorStats getMonitorStats() throws Throwable{
-        if(cachedStats!=null && cachedStats._1 > System.currentTimeMillis()) {
+    MonitorStats getMonitorStats() throws Throwable {
+        if (cachedStats != null && cachedStats._1 > System.currentTimeMillis()) {
```

Comment:
This does not look thread safe as update of the cachedStats occurs at the end of retrieval of the stats multiple threads from threads within that time will all try and retrieve and update the cached value during that window.    Normally we would not expect rapid request to this resource method, but it could happen,  if in a high load condition the stats retrieval is delayed, K8s my retry while still processing and compouning the issue.

My preference would be to not make the call to gather the uncached stats inline with the request to the MonitorResource at all.   The Monitor resource should always be able to return a static (immutable?) object providing the state information that has been updated either on a poll without any extra work, or asynchronously in a background thread.  This way there are no threading issues,  each request should take as long as the next as there is no processing required.
If the background job fails to update, or if the system has hung so it is not able to update the state then this could be a trigger to the liveness probe,  e.g. there is a timestamp in the cached object that is returned to the readiness probe and if it has not been updated in a long time then the Monitor service itself must have failed or the system has hung requiring a restart.

Reply:
We don't need to change to this background service method now if we want to get this out there, but I do want to make sure that multiple requests in a short time cannot cause multiple threads in the uncached process at the same time, only one thread should be allowed to update the cached value until it is fully updated.

Reply:
Don't disagree, I think the only saving grace is that it only gets called on the startup probe.  Once that probe is done the `/alive` and `/ready` only check the cache.

Reply:
should be done.  Added a sync on the method.